src/lib/rule-context.ts

```diff
@@ -107,7 +107,7 @@ export class RuleContext {
         this.sonar.report(
             this.id,
             severity || this.severity,
-            sourceCode,
+            sourceCode || codeSnippet,
```

Comment:
IMO if the `codeSnippet` is present, maybe it should have preference over the `sourceCode`